rhobs/recording/pipelines_availability_recording_rules.yaml

```diff
@@ -0,0 +1,41 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: rhtap-pipelines-availability
+  labels:
+    tenant: rhtap
+
+spec:
+  groups:
+    - name: pipelines_availability
+      interval: 1m
+      rules:
+        # Deployments
+        - record: konflux_up
+          expr: |
+            clamp_max(
+              label_replace(
+                label_replace(
+                  floor(
+                    kube_deployment_status_replicas_available{
+                      namespace=~"openshift-pipelines|tekton-results"
+                    } / clamp_min(kube_deployment_spec_replicas{namespace=~"openshift-pipelines|tekton-results"}, 1)
+                  ), "check", "replicas-available", "",""
+                ), "service", "$1", "deployment","(.*)"
+              ), 1
+            )
+
+        # Stateful Sets
+        - record: konflux_up
+          expr: |
+            clamp_max(
+              label_replace(
+                label_replace(
+                  floor(
+                    kube_statefulset_status_replicas_available{
+                      namespace=~"openshift-pipelines|tekton-results"
+                    } / clamp_min(kube_statefulset_replicas{namespace=~"openshift-pipelines|tekton-results"},1)
+                  ), "check", "replicas-available", "",""
+                ), "service", "$1", "statefulset","(.*)"
+              ), 1
+            )
```

Comment:
These metrics capture every deployment and statefulset, where the specific services which need to be alerted-on will be defined in the alerts. Each of the respective services are in the unit tests. 

Also note that in the event a service's deployment is scaled to 0 the `clamp_min` will ensure the alert fires.